src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -890,6 +891,23 @@ protected virtual Cci.IModuleReference TranslateModule(ModuleSymbol module, Diag
             return namedTypeSymbol;
         }
 
+        private void CheckTupleUnderlying(NamedTypeSymbol namedTypeSymbol, SyntaxNode syntaxNodeOpt, DiagnosticBag diagnostics)
+        {
+            // check that underlying type of a ValueTuple is indeed a value type (or error)
+            // this should never happen, in theory,
+            // but if it does happen we should make it a failure.
+            // NOTE: declaredBase could be null for interfaces
+            var declaredBase = namedTypeSymbol.BaseTypeNoUseSiteDiagnostics;
+            if (declaredBase?.SpecialType != SpecialType.System_ValueType && declaredBase?.IsErrorType() != true)
```

Comment:
It covers the case where ValueTuple is an interface. (and it was indeed causing a crash)

---
In reply to: [81813233](https://github.com/dotnet/roslyn/pull/14161#discussion_r81813233) [](ancestors = 81813233,81812665)